pkg/tbtcpg/moved_funds_sweep.go

```diff
@@ -0,0 +1,403 @@
+			"cannot find moving funds transaction data: [%w]",
+			err,
+		)
+	}
+
+	// Check if the wallet already has a main UTXO.
+	hasMainUtxo := walletChainData.MainUtxoHash != [32]byte{}
+
+	proposal, err := mfst.ProposeMovedFundsSweep(
+		taskLogger,
+		walletPublicKeyHash,
+		movingFundsTxHash,
+		movingFundsOutputIdx,
+		hasMainUtxo,
+		0,
+	)
+	if err != nil {
+		return nil, false, fmt.Errorf(
+			"cannot prepare moved funds sweep proposal: [%w]",
+			err,
+		)
+	}
+
+	return proposal, true, nil
+}
+
+// FindMovingFundsTxData finds the transaction hash and output index for the
+// unswept funds transferred from a source wallet to this wallet. It returns
+// data for only one funds transfer. If there were more than one funds transfers
+// the data for the first encountered unswept transfer is returned.
+func (mfst *MovedFundsSweepTask) FindMovingFundsTxData(
+	walletPublicKeyHash [20]byte,
+) (
+	txHash bitcoin.Hash,
+	txOutputIdx uint32,
+	err error,
+) {
+	blockCounter, err := mfst.chain.BlockCounter()
+	if err != nil {
+		return bitcoin.Hash{}, 0, fmt.Errorf(
+			"failed to get block counter: [%w]",
+			err,
+		)
+	}
+
+	currentBlockNumber, err := blockCounter.CurrentBlock()
+	if err != nil {
+		return bitcoin.Hash{}, 0, fmt.Errorf(
+			"failed to get current block number: [%w]",
+			err,
+		)
+	}
+
+	filterStartBlock := uint64(0)
+	if currentBlockNumber > MovedFundsSweepLookBackBlocks {
+		filterStartBlock = currentBlockNumber - MovedFundsSweepLookBackBlocks
+	}
+
+	// Find all the wallets that recently committed to move funds to this wallet.
+	// The returned data also contains information on the position on the target
+	// wallets list.
+	movingFundsCommitments, err := mfst.findMovingFundsCommitments(
+		walletPublicKeyHash,
+		filterStartBlock,
+	)
+	if err != nil {
+		return bitcoin.Hash{}, 0, fmt.Errorf(
+			"failed to find committing source wallets: [%w]",
+			err,
+		)
+	}
+
+	// Find the transaction outpoints that transfer funds to this wallet coming
+	// from the recent moving funds transactions.
+	movingFundsTxOutpoints, err := mfst.findMovingFundsTxOutpoints(
+		movingFundsCommitments,
+		filterStartBlock,
+	)
+	if err != nil {
+		return bitcoin.Hash{}, 0, fmt.Errorf(
+			"failed to find moving funds transactions: [%w]",
+			err,
+		)
+	}
+
+	// Find the first outpoint that represents an unproven moved funds sweep
+	// request and return it.
+	for _, outpoint := range movingFundsTxOutpoints {
+		movedFundsSweepRequest, err := mfst.chain.GetMovedFundsSweepRequest(
+			outpoint.TransactionHash,
+			outpoint.OutputIndex,
+		)
+		if err != nil {
+			return bitcoin.Hash{}, 0, fmt.Errorf(
+				"failed to get moved funds sweep request: [%w]",
+				err,
+			)
+		}
+
+		if movedFundsSweepRequest.State == tbtc.MovedFundsStatePending {
+			return outpoint.TransactionHash, outpoint.OutputIndex, nil
+		}
+	}
+
+	// No pending moved funds sweep request could be found.
+	return bitcoin.Hash{}, 0, ErrNoPendingMovedFundsSweepRequests
```

Comment:
Are we returning an error because we are invoking `FindMovingFundsTxData` with the assumption that `walletChainData.PendingMovedFundsSweepRequestsCount > 0`?

Reply:
Yes.